
## U.S. INTERNATIONAL TRADE IN GOODS (MARCH 2020)

### The purpose of this project:

This project aims to explore changes in international trade in the U.S. during the Coronavirus outbreak, and thus to further our understanding of the impacts of this epidemic on economic growth.

As we know, a novel virus called COVID-19 began in Wuhan, China in around December 2019; it has spread to almost every country. The virus transmissions have killed hundreds of thousands of people around the world and continue to be one of the most dangerous global pandemics in the 21st century. It has caused many cities and countries in quarantine and leading to countless businesses to lockdown. 

It has been six months since the virus happened in China, but its huge influences are kept continuing - disrupting the global supply chain, business operations, and manufactoring activities which results in the decrease in consumer spending, the collapse of many corporations, and higher unemployment rate.

Although this project only studies how coronavirus negatively impacts economic activities in March 2020, it can give us a better picture of how a novel virus can be a game-changer for global trades and the development of technology in the future.
 


## 1. Guessing game:
### a. Do you know what country has the most goods exported to the U.S. in March 2020?

In [ ]:
import tkinter as tk
from tkinter import Button

root = tk.Tk()

v = tk.IntVar()

v.set(2)  # initializing the choice to Canada

importCountriesGuess = [
    ("China"),
    ("Mexico"),
    ("India"),
    ("Canada"),
    ("Japan")
]

def ShowAnswer():
#     label.config(state="disabled")
    if v.get() == 1:
        print("You are correct!")
    else:
        print("Sorry! Your guess is wrong. Mexico is the correct answer")

tk.Label(root, 
         text="Who is the biggest exporter to the U.S. in March 2020?",
         justify = tk.LEFT,
         padx = 20).pack()

for val, city in enumerate(importCountriesGuess):
    tk.Radiobutton(root, 
                  text=city,
                  padx = 20, 
                  variable=v, 
                  command=ShowAnswer,
                  value=val).pack(anchor=tk.W)

Button(root, text="Quit", command=root.destroy).pack()

root.mainloop()

### b. Do you know what country has the most U.S. goods imported in March 2020?

In [ ]:
import tkinter as tk
from tkinter import Button

root = tk.Tk()

v = tk.IntVar()

v.set(0)  # initializing the choice to Canada

importCountriesGuess = [
    ("China"),
    ("Mexico"),
    ("Germany"),
    ("Canada"),
    ("United Kingdom")
]

def ShowAnswer():
#     label.config(state="disabled")
    if v.get() == 3:
        print("You are correct!")
    else:
        print("Sorry! Your guess is wrong. Canada is the correct answer")

tk.Label(root, 
         text="What country has the most U.S. goods imported in March 2020?",
         justify = tk.LEFT,
         padx = 20).pack()

for val, city in enumerate(importCountriesGuess):
    tk.Radiobutton(root, 
                  text=city,
                  padx = 20, 
                  variable=v, 
                  command=ShowAnswer,
                  value=val).pack(anchor=tk.W)

Button(root, text="Quit", command=root.destroy).pack()

root.mainloop()

## 2. List of Top Trading Partners 

The guessing game above has revealed which country is the biggest importer of U.S. goods and which country is the top exporter to the U.S. The table below shows the list of top 15 U.S trading partners in March 2020 with more information. 

All the [data](https://www.census.gov/foreign-trade/statistics/highlights/top/top2003yr.html
"optional title") was collected by The U.S. Census Bureau and the U.S. Bureau of Economic Analysis

In [3]:
# import all libraries needed

from pandas import DataFrame, read_csv
import matplotlib.pyplot as plt
import pandas as pd 
import numpy as np
import sys #only needed to determine Python version number

# Enable inline plotting
%matplotlib inline

filename = 'InternationalTrade.csv'
data = pd.read_csv(filename)
print(data)

           Country  Exports  Imports  Total Trade Percent of Total Trade
0           Mexico     60.3     87.5        147.8                 15.30%
1           Canada     69.3     74.9        144.2                 14.90%
2            China     22.0     75.9         97.9                 10.10%
3            Japan     18.1     33.3         51.4                  5.30%
4          Germany     15.8     31.0         46.9                  4.90%
5     Korea, South     14.5     18.6         33.1                  3.40%
6   United Kingdom     17.8     14.1         31.9                  3.30%
7            India      8.7     14.0         22.7                  2.40%
8           Taiwan      8.3     13.4         21.7                  2.20%
9           France      9.9     11.5         21.4                  2.20%
10     Netherlands     13.4      7.3         20.7                  2.10%
11         Vietnam      2.9     17.3         20.3                  2.10%
12         Ireland      2.4     17.4         19.8  

## 3. Overview of US International Trade

In [ ]:
import numpy as np
cities = np.array(data.Country)
imports = np.array(data.Imports)
exports = np.array(data.Exports)
plt.plot(cities, imports,
         cities, exports)

plt.xticks(rotation = 90, fontsize = 14)
plt.ylabel('Total Trade in billion dollars', fontsize = 15)

plt.axis(['Mexico', 'Switzerland', 0,100], fontsize = 100)
plt.yticks([0, 20, 40, 60, 80, 100],
           ['0', '20B', '40B', '60B', '80B', '100B'], fontsize = 14)

#make the graph bigger
fig = plt.gcf()
fig.set_size_inches(7, 7)

plt.title("Top 15 trade partners in March 2020", fontsize = 20)
plt.show()

## Analysis:

Looking at the graph above, we can see there are only 2 countries among the top 15 ones, America exported goods to their countries more than they exported goods to the U.S. Those countries are the United Kingdom and the Netherlands. The U.S. got a trade surplus with the U.K and the Netherlands of 3.7 billion and 6.1 billion dollars, respectively. It is surprising that the U.S. obtained benefits from trading with the UK while America is well-known the biggest exporting market from which the UK gained the most through trade for many years. 

However, on the overall, American ran a trade deficit totaling 44.4 billion dollars in 03/2020. Mexico, China, and Canada are still the three earned the most when trading with America. This happened when a dollar has more value compared to other currencies making imported products cheaper, but also making the U.S. exports more expensive for foreign consumers. The trade deficit is unavoidable when the U.S. economy grows - American buyers have a higher income and higher demand to buy more foreign goods. Nevertheless, a large and long-term deficit can lead to bigger national debts, and thus threaten the health of the economy. This is especially important to be considered in the future trade policy when the coronavirus disease is now hitting America more than any other country in the world.

## 4. Top 5 countries partners

In [4]:
cities = np.array(data.Country)
imports = np.array(data.Imports)
exports = np.array(data.Exports)
citiesIm = list(zip(cities, imports))
citiesEx = list(zip(cities, exports))

ExportCities = sorted(citiesEx, key = lambda city: city[1])[::-1]
df1 = pd.DataFrame(data = ExportCities[:5], columns=['COUNTRIES', 'EXPORTS (in billion dollars)'])

df2 = pd.DataFrame(data = citiesIm[:5], columns=['COUNTRIES', 'IMPORTS (in billion dollars)'])

horizontal_stack = pd.concat([df1, df2], axis=1)
horizontal_stack

COUNTRIES  EXPORTS (in billion dollars) COUNTRIES  \
0          Canada                          69.3    Mexico   
1          Mexico                          60.3    Canada   
2           China                          22.0     China   
3           Japan                          18.1     Japan   
4  United Kingdom                          17.8   Germany   

   IMPORTS (in billion dollars)  
0                          87.5  
1                          74.9  
2                          75.9  
3                          33.3  
4                          31.0

### Annotation*:

There is a little note in the table above, as we can see China imported 75.9 billion dollars more than Canada, but it is ranked after Canada. It is because, in total trade, Canada has 144.2 billion dollars with the U.S. more than China, so it is third-ranked after Mexico and Canada on the list. 

## Compare the top five partners in the given list

In [ ]:
#extract only the top 5 cities exporting goods to US
top_5_export_cities = ExportCities[:5]

#convert the list of tuple to an array
cityArray1 = np.asarray(top_5_export_cities)

ci1 = []
ex = []

#using for loop to get an array of city and an array of import
for i in range(len(cityArray1)):
    for j in range(len(cityArray1[i])):
        if j == 0:
            ci1.append(cityArray1[i][j])
        else:
            ex.append(cityArray1[i][j])

cit1 = np.asarray(ci1)
exp = np.asfarray(ex, float)


#extract only the top 5 cities importing goods to US
top_5_import_cities = citiesIm[:5]

#convert the list of tuple to an array
cityArray = np.asarray(top_5_import_cities)

ci = []
im = []

#using for loop to get an array of city and an array of import
for i in range(len(cityArray)):
    for j in range(len(cityArray[i])):
        if j == 0:
            ci.append(cityArray[i][j])
        else:
            im.append(cityArray[i][j])

cit = np.asarray(ci)
imp = np.asfarray(im, float)


#draw 2 pie charts
fig = plt.gcf()
fig.set_size_inches(11,7)

#draw pie 1
plt.subplot(121) # 1 row, 2 colum, 1st chart is on the left
explode = (0.1, 0, 0, 0, 0) 
plt.pie(exp, explode = explode, labels = cit1, autopct='%1.1f%%', shadow=True, startangle=140)
plt.axis('equal') # Equal aspect ratio ensures that pie is drawn as a circle
plt.title("Top 5 countries U.S. exported", fontsize=20)

#draw pie 2
plt.figure(1, figsize=(10, 6))
plt.subplot(122) # 1 row, 2 colum, 2nd chart is on the right
explode = (0.1, 0, 0, 0, 0)  
plt.pie(imp, explode = explode, labels = cit, autopct='%1.1f%%', shadow=True, startangle=140)
plt.axis('equal') # Equal aspect ratio ensures that pie is drawn as a circle
plt.tight_layout()
plt.title("Top 5 countries U.S. imported", fontsize=20)

plt.show()

## 5. Economic and Technology forecasts

As we can see from the graphs above, America imported most of the goods from five countries including China, Mexico, Canada, Japan, and Germany. It is interesting that Mexico was an unexpected winner when replacing China to be the biggest exporters, and Canada even surpassed China to become the second-biggest trading partners with the U.S. The reason is that during February and March, China was struggled desperately to limit new cases and control the epidemic that influenced on manufacturing operations and its supply chain. However, China is now reopening its business and normal life after reporting no new positive cases in many major cities. Therefore, although China was in the third rank, it may again take over their first ranking on the list in the future.

Statistics also presents that Canada and Mexico are the two largest importing market of U.S. goods. Some economic experts even further warn Mexico can become the next China because Mexico is continuing a winner of US-China trade wars and the coronavirus crisis. 

In the context of coronavirus when many startups and small companies are dealing with financial deficits or even filing for bankruptcy; some believe that big tech giants like Amazon, Facebook, or Google with stronger can even become bigger after this pandemic. Nevertheless, there can be a few cases like Zoom, a videoconferencing company, can become the next big tech.

This novel coronavirus is spreading around the world showing higher spending on cybersecurity and the bigger role of AI to prevent the public health crisis. During the outbreak, companies, restaurants, or hospitals need more automated technologies to keep their business but also to minimize face-to-face contact. A robot can be used to diagnose simple illness, deliver medications, or communicate with people that can help government staff, workers, especially doctors and nurses, to limit their potential exposure to infectious patients. This global pandemic will inspire more tech companies and AI experts to innovate in artificial intelligence as it can be the key priority to prepare and tackle the next outbreaks.
